File: components/events/TweetCanceledEvent.ts
import { Environment, fetchQuery, graphql } from "relay-runtime"

import { createEventHook } from "components/EventsProvider"

export interface TweetCanceledEvent {
  user_id: string
  item_id: string
}

const fetchTweetQuery = graphql`
  query TweetCanceledEventQuery($id: ID!) {
    node(id: $id) {
      id
      ... on TweetGroup {
        status
      }
    }
  }
`

export async function handleTweetCanceled(
  environment: Environment,
  event: TweetCanceledEvent
): Promise<void> {
  await fetchQuery(environment, fetchTweetQuery, { id: event.item_id })
}

export const useTweetCanceledEvent = createEventHook(
  "TweetCanceled",
  handleTweetCanceled
)

File: ios/Courier/Courier/Events/TweetCanceledEvent.swift
import Combine
import Relay
import CourierGenerated

struct TweetCanceledEvent: Decodable {
    var userID: String
    var itemID: String

    enum CodingKeys: String, CodingKey {
        case userID = "user_id"
        case itemID = "item_id"
    }
}

private let query = graphql("""
query TweetCanceledEventQuery($id: ID!) {
  tweetGroup(id: $id) {
    id
    status
  }
}
""")

func handleTweetCanceled(_ environment: Environment, _ event: TweetCanceledEvent) -> AnyPublisher<Never, Error> {
    environment.fetchQuery(TweetCanceledEventQuery(variables: .init(id: event.itemID)))
        .ignoreOutput()
        .eraseToAnyPublisher()
}
